### PR TITLE
Optionally set redundancy and contract set in API client

### PR DESCRIPTION
**Pull request by n8maninger**

### CODE REVIEW

File: worker/client.go [hunk 1]
```diff
@@ -217,6 +228,25 @@ func (c *Client) Account(ctx context.Context, hostKey types.PublicKey) (account
 	return
 }
 
+// An APIUploadOption overrides an option on the PUT /objects/*key endpoint.
+type APIUploadOption func(url.Values)
+
+// UploadWithRedundancy sets the min and total shards that should be used for an
+// upload
+func UploadWithRedundancy(minshards, totalshards int) APIUploadOption {
```
Comment (peterjan) [inline worker/client.go:236]: `nit`

```suggestion
func UploadWithRedundancy(minShards, totalShards int) APIUploadOption {
```

Comment (peterjan) [inline worker/client.go:236]: `nit`

```suggestion
func UploadWithRedundancy(minShards, totalShards int) APIUploadOption {
```

File: worker/client.go [hunk 2]
```diff
@@ -217,6 +228,25 @@ func (c *Client) Account(ctx context.Context, hostKey types.PublicKey) (account
...
+	return func(v url.Values) {
+		v.Set(queryStringParamMinShards, strconv.Itoa(minshards))
```
Comment (peterjan) [inline worker/client.go:238]: `nit`

```suggestion
		v.Set(queryStringParamMinShards, strconv.Itoa(minShards))
```

File: worker/client.go [hunk 3]
```diff
@@ -217,6 +228,25 @@ func (c *Client) Account(ctx context.Context, hostKey types.PublicKey) (account
...
+		v.Set(queryStringParamTotalShards, strconv.Itoa(totalshards))
```
Comment (peterjan) [inline worker/client.go:239]: `nit`

```suggestion
		v.Set(queryStringParamTotalShards, strconv.Itoa(totalShards))
```

**Review by peterjan — APPROVED** [review]
pretty neat!

**Review by ChrisSchinnerl — APPROVED** [review]

**Pull request merged**